### PR TITLE
Update force_initdb force_replication and force_efm

### PR DESCRIPTION
**Pull request by hannahms**
Updates and enhances the `force_initdb`, `force_replication` and `force_efm` removal flag tasks. Uninstalls packages, removes configuration files, and reverts configuration settings.

### CODE REVIEW

File: roles/setup_replication/tasks/rm_replication.yml [hunk 1]
```diff
@@ -50,9 +68,54 @@
   become: true
   when: pg_wal|length > 0
 
+- name: Remove postgres default data directory if exists
```
Comment (vibhorkum) [inline roles/setup_replication/tasks/rm_replication.yml:71]: I suggest adding another flag for the removal of the data directory and pg_wal directory, as they contain user data and it is recommended for users to use this flag when deleting these directories. What do you think?

File: roles/setup_replication/tasks/rm_replication.yml [hunk 2]
```diff
@@ -15,9 +16,10 @@
   ansible.builtin.systemd:
     name: "{{ pg_service }}"
     state: stopped
+    enabled: false
   when:
-    - ansible_facts.services[pg_service] is defined
-    - ansible_facts.services[pg_service].state == 'running'
+    - ansible_facts.services[pg_service + '.service'] is defined
```
Comment (vibhorkum) [inline roles/setup_replication/tasks/rm_replication.yml:21]: Do you think we should append the suffix '.service' to the service name specified in the variable, and revise the documentation to explain to users how to utilize the service name variable?

**Review by vibhorkum — CHANGES_REQUESTED** [review]
Could you take a look at the comments? Apart from that, everything seems fine.

File: roles/setup_replication/tasks/rm_replication.yml (hunk 2 above)
Comment (hannahms) [reply roles/setup_replication/tasks/rm_replication.yml:21]: The only place this suffix is necessary is when calling the `ansible_facts.services`, while the `ansible.builtin.systemd` only needs the currently defined `pg_service`. We also call the `pg_service` variable to build other service names so adding the `.service` suffix may require us to remove it in other places.

File: roles/setup_replication/tasks/rm_replication.yml (hunk 1 above)
Comment (hannahms) [reply roles/setup_replication/tasks/rm_replication.yml:71]: I will add a `force_wal` and `force_data` flag

Comment (hannahms) [reply roles/setup_replication/tasks/rm_replication.yml:77]: Should this be added to the tasks removing `pg_default_data` and `pg_local_wal_archive_dir`?

Comment (vibhorkumar123) [reply roles/setup_replication/tasks/rm_replication.yml:77]: The directory is not for WAL archives, but rather for the actual WAL files location in Postgres. The archive location can be configured using archive_command or archive_library, as described in https://www.postgresql.org/docs/current/continuous-archiving.html. As we are currently utilizing the pg_wal and pg_data variables for a user-defined location, let's create new variables with the same name, but prefix with either `rm or `force_rm`

**Review by vibhorkumar123 — APPROVED** [review]
LGTM!

**Pull request merged**